nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/ListGCSBucket.java

```diff
@@ -381,40 +384,36 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 // Update state
                 if (lastModified > maxTimestamp) {
                     maxTimestamp = lastModified;
-                    currentKeys.clear();
+                    maxKeys.clear();
                 }
                 if (lastModified == maxTimestamp) {
-                    currentKeys.add(blob.getName());
+                    maxKeys.add(blob.getName());
                 }
-                listCount++;
+                loadCount++;
             }
 
-            blobPages = blobPages.getNextPage();
-            commit(context, session, listCount);
-            listCount = 0;
-        } while (blobPages != null);
-
-        currentTimestamp = maxTimestamp;
+            commit(context, session, loadCount);
 
-        final long listMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
-        getLogger().info("Successfully listed GCS bucket {} in {} millis", new Object[]{bucket, listMillis});
+            blobPage = blobPage.getNextPage();
+        } while (blobPage != null);
 
-        if (!commit(context, session, listCount)) {
-            if (currentTimestamp > 0) {
-                persistState(context);
-            }
-            getLogger().debug("No new objects in GCS bucket {} to list. Yielding.", new Object[]{bucket});
+        if (maxTimestamp != 0) {
+            currentTimestamp = maxTimestamp;
+            currentKeys = maxKeys;
+            persistState(context);
+        } else {
+            getLogger().debug("No new objects in GCS bucket {} to load. Yielding.", new Object[]{bucket});
             context.yield();
         }
+
+        final long listMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+        getLogger().info("Successfully listed GCS bucket {} in {} millis", new Object[]{bucket, listMillis});
     }
 
-    private boolean commit(final ProcessContext context, final ProcessSession session, int listCount) {
-        boolean willCommit = listCount > 0;
-        if (willCommit) {
-            getLogger().info("Successfully listed {} new files from GCS; routing to success", new Object[] {listCount});
+    private void commit(final ProcessContext context, final ProcessSession session, int loadCount) {
+        if (loadCount > 0) {
+            getLogger().info("Successfully loaded {} new files from GCS; routing to success", new Object[] {loadCount});
```

Comment:
I think 'listing' here is actually more accurate - saying that they were 'loaded' might imply that the contents were transferred from GCS to NiFi.

Reply:
Absolutely right, I will rename it back.

Reply:
Thanks, looks good to me.